app/policies/reservation_policy.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ReservationPolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      if user.manager? || user.employee?
+        scope.all
+      else
+        scope.where(user: user)
+      end
+    end
+  end
+
+  def show?
+    if user.employee? || user.manager?
+      true
+    else
+      user.id == record.user_id
+    end
+  end
+
+  def create?
+    true
+  end
+
+  def update?
+    true
```

Comment:
It would be best to have the same logic here as in show? - you can extract it to some function like `admin_or_owner?` and call it in both.

Reply:
Refactored, thanks :)